test/common/stats/isolated_store_impl_test.cc

```diff
@@ -189,6 +190,41 @@ TEST_F(StatsIsolatedStoreImplTest, AllWithSymbolTable) {
   EXPECT_EQ(2UL, store_->textReadouts().size());
 }
 
+TEST_F(StatsIsolatedStoreImplTest, CounterWithTag) {
+  StatNameTagVector tags{{makeStatName("tag1"), makeStatName("tag1Value")}};
```

Comment:
can you make these tests all include two stats with the same tag-extracted-name but different tags, and make sure they all can be represented?

Reply:
Done (also in the predecessor change that doesn't touch cache, so if you want to merge them separately we can do that).